docs/events.md

```diff
@@ -0,0 +1,123 @@
+## Events
+
+Custom Events provide a way for Apps to validate [message metadata](https://api.slack.com/metadata) against a pre-defined schema.
+
+### Defining a Event
+
+Events can be defined with the top level `DefineEvent` export. Events must be set up as an `object` type or a [`custom Type`][types] of an `object` type. Below is an example of setting up a custom Event that can be used during an incident.
+
+```ts
+const IncidentEvent = DefineEvent({
+  name: "incident",
+  title: "Incident",
+  type: Schema.types.object,
+  properties: {
+    id: { type: Schema.types.string },
+    title: { type: Schema.types.string },
+    summary: { type: Schema.types.string },
+    severity: { type: Schema.types.string },
+    date_created: { type: Schema.types.number },
+  },
+  required: ["id", "title", "summary", "severity"],
+});
+```
+
+Note: By default the validation will not check if additional properties were provided, but this can be done by setting `additionalProperties: false` if this functionality is desired.
+
+### Registering an Event to the App
+
+To register the newly defined event, add it to the array assigned to the `events`
+parameter while defining the [`Manifest`][manifest].
+
+Note: All custom Events **must** be registered to the [Manifest][manifest] in
+order for them to be used. There is no automated registration for Events.
+
+```ts
+Manifest({
+  ...
+  events: [IncidentEvent],
+});
+```
+
+### Referencing Events
+
+There are two places where you can reference your events:
+1. Posting a message to Slack
+2. Creating a Message Metadata Trigger
+
+#### Posting a message to Slack
+Event validation happens against the App's manifest when an App posts a message to Slack using the `metadata` parameter. If the `event_type` matches a custom Event specified in the App's manifest, it will validate that all required parameters are provided. If it doesn't meet the validation standards, a warning will be returned in the response and the metadata will be dropped from the message.
```

Comment:
Not sure where this fits in, but metadata can also be sent/changed when updating a message. The behaviour is otherwise the same as when posting a message.

Reply:
@filmaj I made these updates — it's done against the `name` property